readium/streamer/src/main/java/org/readium/r2/streamer/parser/audio/AudioParser.kt

```diff
@@ -25,18 +28,31 @@ import org.readium.r2.streamer.parser.PublicationParser
  */
 class AudioParser : PublicationParser {
 
-    override suspend fun parse(asset: PublicationAsset, warnings: WarningLogger?): Publication.Builder? {
+    override suspend fun parse(
+        asset: PublicationParser.Asset,
+        warnings: WarningLogger?
+    ): Try<Publication.Builder, PublicationParser.Error> {
 
-        if (!accepts(asset.mediaType, asset.fetcher))
-            return null
+        if (!asset.mediaType.matches(MediaType.ZAB) && !asset.mediaType.isAudio)
+            return Try.failure(PublicationParser.Error.FormatNotSupported())
 
-        val readingOrder = asset.fetcher.links()
-            .filter { link -> with(File(link.href)) { lowercasedExtension in audioExtensions && !isHiddenOrThumbs } }
-            .sortedBy(Link::href)
-            .toMutableList()
+        val readingOrder =
+            if (asset.mediaType.matches(MediaType.ZAB)) {
```

Comment:
Aren't we always in a ZAB case? Unless it's a single audio file, but in this case the sorting doesn't matter.

Reply:
The filtering matters a bit more, though not so much. Anyway, I feel like what we're doing is a bit clearer this way.

Reply:
Fine by me.